docs/static/filebeat-modules.asciidoc

```diff
@@ -0,0 +1,321 @@
+[[filebeat-modules]]
+
+== Working with Filebeat Modules
+
+Starting with version 5.3, Filebeat comes packaged with pre-built 
+{filebeat}filebeat-modules.html[modules] that contain the configuration needed
+to read, parse, and visualize data from various log file formats, such as Nginx,
+Apache2, and MySQL. Each Filebeat module consists of one or more filesets that
+contain ingest node pipelines, Elasticsearch templates, Filebeat prospector
+configurations, and Kibana dashboards.
+
+Filebeat modules do not currently provide Logstash pipeline configurations.
+In the future, Filebeat modules will provide tighter integration with Logstash
+to offer you a more powerful alternative to using ingest node pipelines.
+For now, you can follow the steps in this section to configure Filebeat and
+build Logstash pipeline configurations that are equivalent to the ingest
+node pipelines available with the Filebeat modules.
+
+Then you'll be able to use the sample Kibana dashboards available with Filebeat
+to visualize your data in Kibana.
+
+NOTE: These manual steps will no longer be required when Logstash support
+is added to Filebeat modules in a future release.
+
+To build and run Logstash configurations that provide capabilities similar to
+Filebeat modules:
+
+. Load the Filebeat index pattern and sample Kibana dashboards.
++
+To do this, you need to run Filebeat with the Elasticsearch output enabled and
+specify the `-setup` flag. For example:
++
+[source,shell]
+----------------------------------------------------------------------
+./filebeat -e -setup -E "output.elasticsearch.hosts=["http://localhost:9200"]"
```

Comment:
@suyograo That's how things should work, but that's not how they work right now. In 5.3, if you run a Filebeat module with the logstash output enabled, you see the error `CRIT Exiting: Filebeat modules configured but the Elasticsearch output is not configured/enabled`:

`````
Rhodas-MacBook-Pro:filebeat-5.3.0-darwin-x86_64 dedemorton$ sudo ./filebeat -e -modules=system
2017/03/23 17:03:16.622482 beat.go:285: INFO Home path: [/Users/dedemorton/BuildTesting/5.3.0_e5a8c674/filebeat-5.3.0-darwin-x86_64] Config path: [/Users/dedemorton/BuildTesting/5.3.0_e5a8c674/filebeat-5.3.0-darwin-x86_64] Data path: [/Users/dedemorton/BuildTesting/5.3.0_e5a8c674/filebeat-5.3.0-darwin-x86_64/data] Logs path: [/Users/dedemorton/BuildTesting/5.3.0_e5a8c674/filebeat-5.3.0-darwin-x86_64/logs]
2017/03/23 17:03:16.622515 beat.go:186: INFO Setup Beat: filebeat; Version: 5.3.0
2017/03/23 17:03:16.622554 metrics.go:23: INFO Metrics logging every 30s
2017/03/23 17:03:16.622572 logstash.go:90: INFO Max Retries set to: 3
2017/03/23 17:03:16.622625 outputs.go:108: INFO Activated logstash as output plugin.
2017/03/23 17:03:16.622682 publish.go:295: INFO Publisher name: Rhodas-MacBook-Pro.local
2017/03/23 17:03:16.622771 async.go:63: INFO Flush Interval set to: 1s
2017/03/23 17:03:16.622778 async.go:64: INFO Max Bulk Size set to: 2048
2017/03/23 17:03:16.623475 beat.go:221: INFO filebeat start running.
2017/03/23 17:03:16.623535 metrics.go:51: INFO Total non-zero values: 
2017/03/23 17:03:16.623542 metrics.go:52: INFO Uptime: 4.588856ms
2017/03/23 17:03:16.623546 beat.go:225: INFO filebeat stopped.
2017/03/23 17:03:16.623551 beat.go:339: CRIT Exiting: Filebeat modules configured but the Elasticsearch output is not configured/enabled
Exiting: Filebeat modules configured but the Elasticsearch output is not configured/enabled
`````

That's why I've outlined the steps the way I have. TBH, I think the workflow is kind of jenky overall.